src/de.rs

```diff
@@ -191,6 +191,24 @@ impl<'de, 'a, T: Trait> de::Deserializer<'de> for &'a mut Deserializer<T> {
     }
 }
 
+impl<R> Deserializer<parse::Parser<io::Lines<R>>>
+    where
+    R: io::BufRead,
+{
+    /// Creates an INI deserializer from an `io::BufRead`.
+    pub fn from_bufread(reader: R) -> Self {
+        Deserializer::new(parse::Parser::from_bufread(reader))
+    }
+}
+
+impl<'a> Deserializer<parse::Parser<parse::OkIter<str::Lines<'a>>>>
+{
+    /// Creates an INI deserializer from an `&str`.
+    pub fn from_str<S: AsRef<str> + 'a>(s: &'a S) -> Self {
+        Deserializer::new(parse::Parser::from_str(s))
+    }
+}
+
```

Comment:
Probably the issue here is the type looks wrong and should be changed to `Parser<OkIter<Cloned<str::Lines<'a>>>>` (note the `std::iter::Cloned` addition). That fix needs to be made to `Parser::from_str` as well.

We'll also want to copy [serde_json](https://docs.serde.rs/serde_json/de/index.html)'s module level prototypes (`from_str` will likely need to be `DeserializedOwned` and not `Deserialize<'a>` because the parser does not currently borrow strings).

Reply:
Er I don't know what I was thinking, `Cloned` is not going to help convert a `str` to a `String`. I'll take a look at fixing the `Parser` to accept both types.

Reply:
That fixed it. Yes there are a few more things to implement in this PR, I'll try to add them and ask if I get stuck again somewhere (this is the first time I'm looking into a serde format implementation).